### DISCUSSION TITLE
How to link to libduckdb_static.a?

### DISCUSSION THREAD

**Discussion by edyu** [duckdb/duckdb#9451] [General]
I'm having a whole bunch of `ld.lld: undefined symbol` errors when I'm trying to link a program to the `libduckdb_static.a` under linux. There are things such as `duckdb_re2`, `duckdb::utf8proc_codepoint_to_utf8` all missing.

**Comment by edyu** [comment]
```
error: ld.lld: undefined symbol: std::string duckdb_fmt::v6::internal::grouping_impl<char>(duckdb_fmt::v6::internal::locale_ref)
    note: referenced by ub_duckdb_common.cpp
    note:               ub_duckdb_common.cpp.o:(duckdb_fmt::v6::internal::basic_writer<duckdb_fmt::v6::buffer_range<char>>::int_writer<int, duckdb_fmt::v6::basic_format_specs<char>>::on_num()) in archive libduckdb_static.a
```

**Comment by Maxxen** [comment]
DuckDB build all of its bundled dependencies as separate static libraries, we don't currently bundle everything into one archive (although I think @samansmink is working on changing that). They are called `libduckdb_fsst.a`, `libduckdb_re2.a` etc and should be present in the `third_party` directory of your build directory, or wherever the build installed duckdb. If you link them to in your final program all symbols should be resolved.

**Reply by edyu** [reply]
Thank you, I assume the `third_party` is something I have to build myself? The reason I'm asking is that `libduckdb_static.a` is now finally bundled as part of the release.

**Reply by Maxxen** [reply]
Yeah. Hmm, that seems like an oversight on our part. I think bundling everything is something we are working on, but could you open an issue so we can track this?

**Reply by edyu** [reply]
https://github.com/duckdb/duckdb/issues/9475 filed as you suggested.

**Reply by paultiq** [reply]
I'm a bit confused by this: `libduckdb_static.a` is bundled as part of the https://duckdb.org/install/?platform=linux&environment=c download, but the same problem exists. 

Is libduckdb_static.a supposed to be linkable now, or should it have been removed from the releases?
